### PR TITLE
25 param

### LINKED ISSUE #25: QLibrary Tree Widget
### What is the feature being requested?
Allow creation of new qcomponents via a drop-down tree menu with all available components for the user to pick from (transmon pocket, CPW, etc.), preferably with images of the component itself.

### What are use cases for this feature?
User can create components by selecting them through a dropdown rather than typing python code

### PR DESCRIPTION
**Pull request by grace-harper-ibm**
### What are the issues this pull addresses (issue numbers / links)?
Closes #25

### Did you add tests to cover your changes (yes/no)?
yes
### Did you update the documentation accordingly (yes/no)?
yes
### Did you read the CONTRIBUTING document (yes/no)?
yes
### Summary
Users can now create QComponents directly from the GUI

### CODE REVIEW

File: qiskit_metal/_gui/widgets/library_new_qcomponent/parameter_entry_window.py [hunk 1]
```diff
@@ -289,12 +289,18 @@ def generate_model_data(self):
                     param_dict[param.name] = create_default_from_type(
                         param.annotation)
 
-        if 'default_options' in self.qcomp_class.__dict__:
-            options = self.qcomp_class.default_options
+        try:
+            print("getting template options")
+            options = self.qcomp_class.get_template_options(self._design)
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/_gui/widgets/library_new_qcomponent/parameter_entry_window.py:294]: Nice

**Comment by priti-ashvin-shah-ibm** [comment]
Observation: Presently, there is just one QComponent in module.  Just want to confirm we will not have multiple QComponents in within one module?

**Comment by priti-ashvin-shah-ibm** [comment]
image
Is this an easy change, or can I request for future issue to be added?

**Review by priti-ashvin-shah-ibm — APPROVED** [review]
Ran a notebook and added QComponents through GUI. Was able to work with both interactively. Really Great work.  !!!

Noticed on command line that every test in run_all_tests failed.  Seems to be reflected in CI also.

**Review by JeremyDrysdale-ibm — CHANGES_REQUESTED** [review]
Please merge in main, and fix the merge conflicts, then I'll review.

**Review by JeremyDrysdale-ibm — APPROVED** [review]
LGTM

**Comment by priti-ashvin-shah-ibm** [comment]
Don't want to hold up the pr.  It can be a separate issue.
Is it possible to move the dividing bar?  Maybe user can “doc” the window when working on entering the data?

**Review by priti-ashvin-shah-ibm — APPROVED** [review]
I have made two requests.  If they are not quick fixes, please make them as new issues. Will approve.  This is a big, very nice feature.  Great job.

**Comment by JeremyDrysdale-ibm** [comment]
> Don't want to hold up the pr. It can be a separate issue.
> Is it possible to move the dividing bar? Maybe user can “doc” the window when working on entering the data?

@priti-ashvin-shah-ibm Perhaps that's best as a separate issue.  This PR already contains a LOT of completed work.

**Review by priti-ashvin-shah-ibm — APPROVED** [review]

**Review by JeremyDrysdale-ibm — APPROVED** [review]
LGTM

**Review by ThomasGM4 — APPROVED** [review]
Looks good.
Still some discussion on how to best manage variable aspects (such as connection pads for qubits).

**Pull request merged**